examples/pydantic/worker.py

```diff
@@ -0,0 +1,84 @@
+from typing import cast
+
+from dotenv import load_dotenv
+from pydantic import BaseModel
+
+from hatchet_sdk import Context, Hatchet
+
+load_dotenv()
+
+hatchet = Hatchet(debug=True)
+
+
+class ParentInput(BaseModel):
+    x: str
+
+
+def pretty_print(tag: str, model: BaseModel) -> None:
+    print()
+    print(tag, model, "\nType:", type(model))  ## This is an instance of `ChildInput`
+    print()
+
+
+@hatchet.workflow(input_validator=ParentInput)
+class Parent:
+    @hatchet.step(timeout="5m")
+    async def spawn(self, context: Context):
+        ## Cast your `workflow_input` to the type of your `input_validator`
+        input = cast(ParentInput, context.workflow_input())
+
+        pretty_print("Workflow Input:", input)  ## This is a `ParentInput`
+
+        child = await context.aio.spawn_workflow(
+            "Child",
+            {"a": 1, "b": "10"},
+        )
+
+        return await child.result()
+
+
+class ChildInput(BaseModel):
+    a: int
+    b: int
+
+
+class StepResponse(BaseModel):
+    status: str
+
+
+@hatchet.workflow()
```

Comment:
Should this workflow have an `input_validator` set?

Reply:
nice catch, thanks!